src/main/java/seedu/address/logic/commands/EditDoctorCommand.java

```diff
@@ -0,0 +1,239 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SPECIALITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_YOE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_DOCTORS;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Doctor;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Specialty;
+import seedu.address.model.person.Yoe;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits the details of an existing doctor in the address book.
+ */
+public class EditDoctorCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit-doc";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the doctor identified "
+            + "by the index number used in the displayed doctors list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_SPECIALITY + "SPECIALITY "
```

Comment:
```suggestion
+ PREFIX_SPECIALTY + "SPECIALTY"
```

Standardise spelling across Docedex and use US English standard.